frontend/public/components/hpa.tsx

```diff
@@ -34,64 +38,61 @@ const MetricsRow: React.FC<MetricsRowProps> = ({ type, current, target }) => (
 
 const externalRow = (metric, current, key) => {
   const { external } = metric;
-  const type = external.metricName;
-  // TODO: show metric selector for external metrics?
-  const currentValue = external.targetAverageValue
-    ? _.get(current, 'object.currentAverageValue')
-    : _.get(current, 'object.currentValue');
-  const targetValue = external.targetAverageValue
-    ? external.targetAverageValue
-    : external.targetValue;
+  const type = external.metric.name;
+  const currentValue =
+    current?.external?.current?.averageValue || current?.external?.current?.value;
+  const targetValue = external.target.averageValue || external.target.value;
 
   return <MetricsRow key={key} type={type} current={currentValue} target={targetValue} />;
 };
 
-const objectRow = (metric, current, ns, key) => {
+const objectRow = (metric, current, ns, key, scaleTarget) => {
   const { object } = metric;
   const type = (
     <>
-      {object.metricName} on
+      {object?.metric?.name} on
       <ResourceLink
-        kind={object.target.kind}
-        name={object.target.name}
+        kind={scaleTarget.kind}
+        name={scaleTarget?.name}
```

Comment:
We should either use optional chaining on both of these lines or neither. `scaleTargetRef` is required, so optional chaining shouldn't be needed.

```suggestion
        kind={scaleTarget.kind}
        name={scaleTarget.name}
```